dear_petition/petition/etl/load.py

```diff
@@ -93,3 +92,42 @@ def create_documents(petition):
         )
         attachment.offense_records.add(*attachment_records)
         previous_document = attachment
+
+
+def assign_agencies_to_documents(petition):
+    agencies = petition.agencies.all()
+    current_document = None
+    i = 0
+    while True:
+        current_document_agencies = agencies[
+            i : (i + 3)
+        ]  # 3 boxes for agencies per document
+        if not current_document_agencies:
+            break
+
+        if not current_document:
+            current_document = petition.base_document
+        else:
+            try:
+                current_document = current_document.following_document
+            except pm.PetitionDocument.DoesNotExist:  # Have to create new attachment
+                current_document = pm.PetitionDocument.objects.create(
+                    petition=petition, previous_document=current_document
+                )
+
+        current_document.agencies.clear()
+
+        for agency in current_document_agencies:
+            current_document.agencies.add(agency)
+
+        i += 3
+
+    try:
+        current_document = current_document.following_document
+        if not current_document.offense_records.exists():
+            # This must have been an extra attachment for agencies, but agencies were removed so no longer necessary
+            current_document.delete()
+    except:
+        pass
+
+    return petition
```

Comment:
This does not handle edge case where an existing attachment needs to have it's agency cleared